toffy/mph_comp_test.py

```diff
@@ -0,0 +1,142 @@
+
+    good_path = os.path.join(Path(__file__).parent, "data", "tissue")
+    good_fov = 'fov-1-scan-1'
+
+    # bad directory path should raise an error
+    with pytest.raises(ValueError):
+        mph.get_estimated_time(bad_path, good_fov)
+
+    # bad fov name data should raise an error
+    with pytest.raises(FileNotFoundError):
+        mph.get_estimated_time(good_path, bad_fov)
+
+    # bad FOV json file data should raise an error, no frameSize or dwellTimeMillis keys
+    bad_data = {"fov": {"not_frameSize": 0, "not_dwellTimeMillis": 0}}
+    with tempfile.TemporaryDirectory() as tmpdir:
+        with open(os.path.join(tmpdir, 'fov_name.json'), 'w') as f:
+            json.dump(bad_data, f)
+        with pytest.raises(KeyError, match="missing one of the necessary keys"):
+            mph.get_estimated_time(tmpdir, 'fov_name')
+
+    # test successful time data retrieval
+    assert mph.get_estimated_time(good_path, good_fov) == 512
+
+
+def test_compute_mph_metrics():
+    bin_file_path = os.path.join(Path(__file__).parent, "data", "tissue")
+    fov_name = 'fov-1-scan-1'
+    target_name = 'CD8'
+    start_mass = -0.3
+    stop_mass = 0.0
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        # bad directory path should raise an error
+        bad_path = os.path.join(Path(__file__).parent, "data", "not-a-folder")
+        with pytest.raises(ValueError):
+            mph.compute_mph_metrics(bad_path, tmpdir, fov_name,
+                                    target_name, start_mass, stop_mass)
+
+        # invalid fov name should raise an error
+        with pytest.raises(FileNotFoundError):
+            mph.compute_mph_metrics(bin_file_path, tmpdir, "not-a-fov",
+                                    target_name, start_mass, stop_mass)
+
+        # invalid target name should raise an error
+        with pytest.raises(ValueError, match="target name is invalid"):
+            mph.compute_mph_metrics(bin_file_path, tmpdir, fov_name,
+                                    "not-a-target", start_mass, stop_mass)
+
+        # test successful data retrieval and csv output
+        mph.compute_mph_metrics(bin_file_path, tmpdir, fov_name,
+                                target_name, start_mass, stop_mass)
+        csv_path = os.path.join(tmpdir, fov_name + '-pulse_height.csv')
+        assert os.path.exists(csv_path)
+
+        # check the csv data is correct
+        mph_data = pd.DataFrame([{
+                'fov': fov_name,
+                'MPH': 2222,
+                'total_count': 72060,
+                'time': 512,
+            }])
+        csv_data = pd.read_csv(csv_path)
+        assert csv_data.equals(mph_data)
+
+
+def test_combine_mph_metrics():
+    bad_path = os.path.join(Path(__file__).parent, "data", "not-a-folder")
+
+    # bad directory path should raise an error
+    with pytest.raises(ValueError):
+        mph.combine_mph_metrics(bad_path)
+
+    data1 = pd.DataFrame([{
+        'fov': 'fov-1-scan-1',
```

Comment:
Since this is fake data, it can be made programmatically rather than harcoding. Also, it can be easier numbers to keep track of mentally